app/src/test/java/org/simple/clinic/appupdate/CheckAppUpdateAvailabilityTest.kt

```diff
@@ -0,0 +1,75 @@
+package org.simple.clinic.appupdate
+
+import android.app.Application
+import android.app.PendingIntent
+import com.google.android.play.core.appupdate.AppUpdateInfo
+import com.google.android.play.core.install.model.InstallStatus
+import com.google.android.play.core.install.model.UpdateAvailability
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.whenever
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(JUnitParamsRunner::class)
+class CheckAppUpdateAvailabilityTest {
+
+  private val config = mock<AppUpdateConfig>()
+
+  lateinit var checkUpdateAvailable: CheckAppUpdateAvailability
+
+  private val currentAppVersionCode = 1
+  private val diffToShowUpdate = 1
+
+  @Before
+  fun setup() {
+
+    val versionCodeCheck = { versionCode: Int, _: Application, _: AppUpdateConfig ->
+      versionCode.minus(currentAppVersionCode) >= diffToShowUpdate
+    }
+
+    checkUpdateAvailable = CheckAppUpdateAvailability(mock(), config, versionCodeCheck)
+  }
+
+  @Test
+  @Parameters(method = "params for checking app update")
+  fun `when app update is available and version is applicable, then shouldShowUpdateDialog should return true`(
+      availableVersionCode: Int,
+      updateAvailabilityState: Int,
+      flexibleUpdateIntent: PendingIntent?,
```

Comment:
Is there a need to pass this as a parameter since we aren't even verifying anything on it? We can just inline the `mock<PendingIntent>()` when constructing the `AppUpdateInfo` instance.

Reply:
We are actually verifying. If we pass a `null` here, it should fail.

Reply:
Oh right, there was one which was `null`, which I missed.